village/src/App.js

```diff
@@ -19,6 +35,9 @@ class App extends Component {
       <div className="App">
         <SmurfForm />
         <Smurfs smurfs={this.state.smurfs} />
+        <Route exact path="/smurfs" render={props => <Smurfs {...props} smurfs={this.state.smurfs} />} />
+          <Route path="/smurfs/:id" render={props => <SmurfForm {...props} smurfs={this.state.smurfs} />} />
```

Comment:
here you seem to be rendering SmurfForm twice, once on link 36, and another time on link 39 inside of the route.